### DISCUSSION TITLE
What summarization method is used in geopandas.GeoDataFrame.plot for multiple values?

### DISCUSSION THREAD

**Discussion by lcoandrade** [geopandas/geopandas#2980] [Q&A] [answered]
I was making a distance analysis using geopandas using a hexagon planar subdivision of a region of interest (hexagons_gdf) and points of interest (poi_gdf).

The first thing I did was to calculate the distances between the hexagons centroids and the points.

With this I created a heatmap_data like this:
```
# Create a new DataFrame for the heatmap data
heatmap_data = pd.DataFrame(distances_array, columns=poi_gdf.index)

# Combine with hexagon geometries
heatmap_data['geometry'] = hexagons_gdf.geometry

#Creating a one-to-one relation in our data set, 
#generating a line for each hexagon-POI relation keeping the distances in a separete column    
heatmap_data = heatmap_data.melt(id_vars='geometry', var_name='point_id', value_name='distances')

# Convert back to GeoDataFrame
heatmap_data = gpd.GeoDataFrame(heatmap_data, geometry='geometry')
```

Then, I plot this heatmap_data with this code:
```
# Plot the georeferenced heatmap
fig, ax = plt.subplots(figsize=(10, 8))
heatmap_data.plot(column='distances', cmap='viridis', legend=True, ax=ax)

# Customize the plot
ax.set_title('Georeferenced Heatmap: Distance Analysis')
ax.set_xlabel('Longitude')
ax.set_ylabel('Latitude')

# Plot points of interest on top of the heatmap
poi_gdf.plot(ax=ax, color='red', markersize=50, label='Points of interest')

plt.legend()
plt.show()
```

So, I was wondering how geopandas determine the value to be used in the plot as heatmap_data has many distance values for each hexagon. Is it the mean value? Where can I find this?

Thanks!

**Comment by m-richards** [comment, marked as answer]
Hi @lcoandrade, thanks for the question. I hope I'm understanding it right based on the snippets you've shown, but no summation is done when plotting - all duplicates are layed on top, with the last record shown on top. This might help illustrate:
```python
from geodatasets import get_path
import pandas as pd
import geopandas as gpd
gdf = gpd.read_file(get_path('nybb'))
gdf2 = pd.concat([gdf.iloc[[0]].assign(distance=i) for i in range(5)])
gdf2.plot(column='distance', cmap='viridis', legend=True)
gdf2.sort_values(by='distance', ascending=False).plot(column='distance', cmap='viridis', legend=True)
```


So you would need to take care of duplication / aggregation yourself before plotting.

**Reply by lcoandrade** [reply]
Thanks for your answer! Now, I can be more sure of what I'm doing. I've decided to calculate the mean distances for each hexagon before plotting. I even made a function with different aggregation functions to give the user more control over the hetman plotting.

Thanks again!!!!
